cognite/neat/rules/models/_rules/_validators/_base.py

```diff
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+
+from cognite.neat.rules.issues import IssueList
+from cognite.neat.rules.models._rules import DMSRules, DomainRules, InformationRules
+
+
+class BaseRulesValidator(ABC):
```

Comment:
Good case for generics
```suggestion
T_Rule = TypeVar("T_Rule", bound=Rules)


class BaseRulesValidator(ABC, Generic[T_Rule]):
```